src/vma/dev/ring.h

```diff
@@ -121,47 +121,43 @@ class ring
 	virtual bool		is_ratelimit_supported(struct vma_rate_limit_t &rate_limit) = 0;
 	virtual int		reg_mr(void *addr, size_t length, uint32_t &lkey) { NOT_IN_USE(addr); NOT_IN_USE(length); NOT_IN_USE(lkey); return -1;};
 	virtual int		dereg_mr(void *addr, size_t length) { NOT_IN_USE(addr);NOT_IN_USE(length); return -1;};
-#ifdef DEFINED_SOCKETXTREME		
+
 	virtual int		socketxtreme_poll(struct vma_completion_t *vma_completions, unsigned int ncompletions, int flags) = 0;
-	virtual int		socketxtreme_reclaim_single_recv_buffer(mem_buf_desc_t* rx_reuse_lst) {NOT_IN_USE(rx_reuse_lst); return -1;}
-	virtual void		socketxtreme_reclaim_recv_buffers(mem_buf_desc_t* rx_reuse_lst) {NOT_IN_USE(rx_reuse_lst); return;}
 
-	inline void set_vma_active(bool flag) {m_vma_active = flag;}
-	inline bool get_vma_active(void) {return m_vma_active;}
+	inline bool is_socketxtreme(void) {return m_socketxtreme.active;}
 
 	inline void put_ec(struct ring_ec *ec)
 	{
-		m_lock_ec_list.lock();
-		list_add_tail(&ec->list, &m_ec_list);
-		m_lock_ec_list.unlock();
+		m_socketxtreme.lock_ec_list.lock();
+		list_add_tail(&ec->list, &m_socketxtreme.ec_list);
+		m_socketxtreme.lock_ec_list.unlock();
 	}
 
 	inline void del_ec(struct ring_ec *ec)
 	{
-		m_lock_ec_list.lock();
+		m_socketxtreme.lock_ec_list.lock();
```

Comment:
I dont understand why socketxtreme logic is located here.
Why not in ring_simple ?

Reply:
to be ready for **ring_bond** usage

Reply:
this is not related, i can agree with it for the api (functions.) but this is socketxtreme internal code which must be located in ring simple.